unit_testing/test_labels.py

```diff
@@ -216,3 +216,17 @@ def test_remove_other_labels_from_image(test_image):
     diff = res.data == expected.data
 
     assert diff.all()
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_find_missing_label(test_image):
+    img = test_image.copy()
+    false_positive = test_image.copy()
+
+    # modifying the data to create one false negative and one false positive
+    # label 1500 is not in test_image originally, and label 111 is.
+    false_positive.data = np.where(false_positive.data == 111, 1500, false_positive.data)
+    FP, FN = sct_labels.find_missing_label(false_positive, img)
+
+    assert int(FP[0]) == 1500
+    assert int(FN[0]) == 111
```

Comment:
I think it might be better to make this an test for `compute_mean_square_error` instead. The context for this bug is because of `sct_label_utils -i <image> -MSE <REF>` didn't work right (from #2986). So, I think the test should match the actual real-world usage as closely as possible to verify that things are working as expected. e.g.

* Running MSE with images that have mismatched labels -> asserting that a warning is thrown
* Running MSE with images that have matching labels -> asserting that no warning is thrown

Reply:
I will add a test for that.

Reply:
in 08fc3e4, I added a test to check if the warning is here when needed.